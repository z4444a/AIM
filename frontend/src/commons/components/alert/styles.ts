import { Theme } from '@material-ui/core/styles/createMuiTheme';
import createStyles from '@material-ui/core/styles/createStyles';
import { WithStyles } from '@material-ui/core/styles/withStyles';
import { amber, green } from '@material-ui/core/colors';

export const styles = (theme: Theme) =>
  createStyles({
    container: {
      width: '100%',
    },
    success: {
      backgroundColor: green[600],
    },
    error: {
      backgroundColor: theme.palette.error.light,
    },
    info: {
      backgroundColor: theme.palette.primary.dark,
    },
    warning: {
      backgroundColor: amber[700],
    },
    icon: {
      fontSize: 20,
    },
    iconVariant: {
      opacity: 0.9,
      marginRight: theme.spacing.unit,
    },
    message: {
      display: 'flex',
      alignItems: 'center',
    },
  });

export type Style = WithStyles<
  'success' | 'error' | 'info' | 'warning' | 'icon' | 'iconVariant' | 'message' | 'container'
>;
