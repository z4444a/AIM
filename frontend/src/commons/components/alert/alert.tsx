import React, { ReactNode } from 'react';
import SnackbarContent from '@material-ui/core/SnackbarContent';
import CheckCircle from '@material-ui/icons/CheckCircle';
import Error from '@material-ui/icons/Error';
import Info from '@material-ui/icons/Info';
import Warning from '@material-ui/icons/Warning';
import { Style } from './styles';

const variantIcon = {
  success: CheckCircle,
  warning: Warning,
  error: Error,
  info: Info,
};

export interface Props {
  message?: string;
  type: AlertType;
}

export type AlertType = 'success' | 'warning' | 'error' | 'info';
export type InternalProps = Style & Props;

export class Alert extends React.PureComponent<InternalProps> {
  public render(): ReactNode {
    const { classes, type, message } = this.props;
    const Icon = variantIcon[type];
    return (
      <SnackbarContent
        className={`${classes[type]} ${classes.container}`}
        aria-describedby="client-snackbar"
        message={
          <span id="client-snackbar" className={classes.message}>
            <Icon className={`${classes.icon} ${classes.iconVariant}`} />
            {message}
          </span>
        }
      />
    );
  }
}
