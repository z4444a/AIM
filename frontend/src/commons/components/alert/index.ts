import withStyles from '@material-ui/core/styles/withStyles';
import { Alert } from './alert';
import { styles } from './styles';

export default withStyles(styles)(Alert);
