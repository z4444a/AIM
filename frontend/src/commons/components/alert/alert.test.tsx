import { mount, shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';
import { Alert, AlertType, InternalProps } from './alert';
import CheckCircle from '@material-ui/icons/CheckCircle';
import Error from '@material-ui/icons/Error';
import Info from '@material-ui/icons/Info';
import Warning from '@material-ui/icons/Warning';

describe('Alert', () => {
  function getComponentProps(type: AlertType): InternalProps {
    const props: InternalProps = {
      classes: {
        container: 'container',
        error: 'error',
        icon: 'icon',
        iconVariant: 'iconVariant',
        info: 'info',
        message: 'message',
        success: 'success',
        warning: 'warning',
      },
      message: 'Message',
      type,
    };
    return props;
  }

  function generateShallowWrapper(type: AlertType) {
    return shallow(<Alert {...getComponentProps(type)} />);
  }

  function generateMountWrapper(type: AlertType) {
    return mount(<Alert {...getComponentProps(type)} />);
  }

  it('should render correctly', () => {
    const wrapper = generateShallowWrapper('info');
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('should render error icon', () => {
    const wrapper = generateMountWrapper('error');
    expect(wrapper.find(Error).exists()).toBe(true);
  });

  it('should render info icon', () => {
    const wrapper = generateMountWrapper('info');
    expect(wrapper.find(Info).exists()).toBe(true);
  });

  it('should render success icon', () => {
    const wrapper = generateMountWrapper('success');
    expect(wrapper.find(CheckCircle).exists()).toBe(true);
  });

  it('should render warning icon', () => {
    const wrapper = generateMountWrapper('warning');
    expect(wrapper.find(Warning).exists()).toBe(true);
  });
});
