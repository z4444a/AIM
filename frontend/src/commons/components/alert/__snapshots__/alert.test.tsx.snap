// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert should render correctly 1`] = `
<WithStyles(SnackbarContent)
  aria-describedby="client-snackbar"
  className="info container"
  message={
    <span
      className="message"
      id="client-snackbar"
    >
      <pure(InfoIcon)
        className="icon iconVariant"
      />
      Message
    </span>
  }
/>
`;
